### PR TITLE
feat: build quad-tree-split view

### PR DESCRIPTION
**Pull request by seanwu1105**
Preview: https://i.imgur.com/EWJ4fS5.mp4

Press up arrow key to split the view and down arrow key to revert the operation.

We might want to change the texture of the wall to make it less visually disturbing.

### CODE REVIEW

**Review by RaymondHuang210129 — CHANGES_REQUESTED** [review]

File: src/quad-tree.ixx [hunk 1]
```diff
@@ -0,0 +1,67 @@
+module;
+
```
[the remark this answers is not in the document]
Comment (seanwu1105) [reply src/quad-tree.ixx:2]: I cannot get what you are trying to mention here. Could you elaborate it more?

[the remark this answers is not in the document]
Comment (RaymondHuang210129) [reply src/quad-tree.ixx:2]: Seems to be solved, thanks!

**Review by RaymondHuang210129 — APPROVED** [review]
LGTM

**Review by sdCaramel302812 — APPROVED** [review]
lgtm

**Review by JoannaWang1216 — APPROVED** [review]
LGTM

**Pull request merged**